app/frontend/src/features/profile/useUpdateHostingPreferences.test.tsx

```diff
@@ -0,0 +1,125 @@
+import { renderHook } from "@testing-library/react-hooks";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { act } from "react-test-renderer";
+import { service } from "../../service";
+import { addDefaultUser } from "../../test/utils";
+import useAuthStore from "../auth/useAuthStore";
+import useCurrentUser from "../userQueries/useCurrentUser";
+import useUpdateHostingPreferences from "./useUpdateHostingPreferences";
+import wrapper from "../../test/hookWrapper";
+
+const getUserMock = service.user.getUser as jest.Mock;
+const updateHostingPreferenceMock = service.user
+  .updateHostingPreference as jest.Mock;
+
+beforeAll(() => {
+  // Mock out console.error so the test output is less noisy when
+  // an error is intentionally thrown for negative tests
+  jest.spyOn(console, "error").mockReturnValue(undefined);
```

Comment:
This also hides a bunch of warnings about not using "act", which seems like a `@testing-library/react-hooks` bug?

Reply:
That usually happens either when something is still loading and could change some component state and we're not waiting for it, or whatever that could change some component state is not wrapped in `act`. My guess would be the first option here.

Maybe it's not such a good idea to silence errors by mocking `console.error` out completely like that after all. I rather have some noise and be able to see the errors.

Reply:
Is there a way around it? React-query does so much stuff in the background...

Reply:
I have some similar tests where I only cared about the loading state and calling `unmount` from the destructured result of `renderHook` at the end of the test fixed my warnings! I think that's probably the way to go as it looks like React Query [cleans up properly by unmounting the client here](https://github.com/tannerlinsley/react-query/blob/master/src/react/QueryClientProvider.tsx#L41) when the React tree gets unmounted by calling `unmount` from hooks testing lib. Give that a try?